### ISSUE TITLE
[DOC] List Grafana Agent features and then move feature doc to Agent doc; remove doc in Tempo

### ISSUE THREAD

**Issue by knylander-grafana** [grafana/tempo#2330]
The Tempo documentation for Grafana Agent is out of date. Update the Grafana Agent docs in Tempo: 

* Move as much config and set up info as possible from Grafana Agent doc in Tempo repo to Agent doc 
* Summarize the tracing-related features in the Agent 
* Point people to the actual Agent documentation: https://grafana.com/docs/agent/latest/
* Unpublish the tail-based-sampling doc. 

Relates to: https://github.com/grafana/tempo/issues/2250 and https://github.com/grafana/tempo/issues/2305

**Comment by knylander-grafana** [comment]
@clayton-cornell FYI.

**Comment by knylander-grafana** [comment]
@clayton-cornell With the focus now on Agent Flow, we'll need to decide what the best thing to do with the Agent pages that we have in Tempo docs. These pages explain the tracing-related features and then give configuration examples.

**Comment by clayton-cornell** [comment]
To keep things in sync, can we mount the Agent docs in Tempo like we do for Cloud? Or are there one-off Tempo-specific topics that don't overlap?

**Comment by knylander-grafana** [comment]
@clayton-cornell I don't think we need to mount the Agent docs in Tempo; rather we want to move the Agent-related docs in Tempo to Agent docs instead. See these page: https://grafana.com/docs/tempo/latest/configuration/grafana-agent/ 

We have content in these docs about Agent capabilities pertinent to tracing data. I haven't seen any equivalent content in the Agent docs per se. It would be great if ew could incorporate that content inot the Agent docs. It doesn't make sense to have them only in Tempo. People coming to the Agent docs wouldn't know about them. (Plus the Agent content in Tempo gets out of date.)

**Comment by knylander-grafana** [comment]
Updated this doc issue for Alloy instead of Agent.

**Comment by knylander-grafana** [comment]
We'll come back to this if it's important. Closing becuase it's way out of date.
